clang/include/clang/Driver/Options.td

```diff
@@ -6911,6 +6911,18 @@ def fsycl_fp32_prec_sqrt : Flag<["-"], "fsycl-fp32-prec-sqrt">, Group<sycl_Group
   Visibility<[ClangOption, CC1Option]>,
   HelpText<"SYCL only. Specify that single precision floating-point sqrt is correctly rounded.">,
   MarshallingInfoFlag<CodeGenOpts<"SYCLFp32PrecSqrt">>;
+def fsycl_use_integration_headers : 
+  Flag<["-"], "fsycl-use-integration-headers">, Flags<[HelpHidden]>,
+  Group<sycl_Group>, Visibility<[ClangOption, CLOption, CC1Option]>,
+  HelpText<"SYCL only. Specify that integration headers be used for "
+           "communication between the host and the device compilations.">,
+  MarshallingInfoFlag<LangOpts<"SYCLUseIntegrationHeaders">>;
+def fno_sycl_use_integration_headers : 
+  Flag<["-"], "fno-sycl-use-integration-headers">, Flags<[HelpHidden]>,
+  Group<sycl_Group>, Visibility<[ClangOption, CLOption, CC1Option]>,
+  HelpText<"SYCL only. Specify that integration headers not be used for "
+           "communication between the host and the device compilations.">,
+  MarshallingInfoNegativeFlag<LangOpts<"SYCLUseIntegrationHeaders">>;
```

Comment:
Is it possible to combine these into a single positive/negative form? Something like:
```
def fsycl_use_integration_headers : BoolFOption<"sycl-use-integration-headers",
  DefaultTrue,
  PosFlag<SetTrue>, NegFlag<SetFalse>,
  BothFlags<[HelpHidden], [ClangOption, CLOption, CC1Option]>,
  Group<sycl_Group>,
  HelpText<"SYCL only. Specify that integration headers be used for "
           "communication between the host and the device compilations.">,
  MarshallingInfoFlag<LangOpts<"SYCLUseIntegrationHeaders">>;
```

Reply:
I tried, but couldn't.   There is a limit of 7 template arguments for BoolFOption and I couldn't figure out how to retain all that info.

@mdtoguchi, can you help here?   Is this possible to do?